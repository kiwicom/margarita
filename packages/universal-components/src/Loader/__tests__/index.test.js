// @flow

import * as React from 'react';
import { Platform } from 'react-native';
import { render } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';

import { Loader, PageLoader } from '../index';

jest.mock('Platform', () => ({
  select: jest.fn(),
}));

describe('Loader', () => {
  it('should match snapshot diff between small and large loader', () => {
    const small = render(<Loader size="small" />);
    const large = render(<Loader size="large" />);

    expect(snapshotDiff(small, large, { contextLines: 1 })).toMatchSnapshot();
  });
});

describe('Page Loader', () => {
  it('should have large page loader for Android', () => {
    Platform.select.mockImplementationOnce(obj => obj.android);
    const { getByType } = render(<PageLoader />);

    expect(getByType(Loader).props.size).toBe('large');
  });

  it('should have small page loader for iOS', () => {
    Platform.select.mockImplementationOnce(obj => obj.ios);
    const { getByType } = render(<PageLoader />);

    expect(getByType(Loader).props.size).toBe('small');
  });
});
