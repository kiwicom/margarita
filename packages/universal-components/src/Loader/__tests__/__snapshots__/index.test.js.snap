// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot diff between small and large loader 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -4,3 +4,3 @@
    hidesWhenStopped={true}
-   size=\\"small\\"
+   size=\\"large\\"
  />"
`;
